packages/pluggableWidgets/tree-view-web/src/components/TreeView.tsx

```diff
@@ -152,13 +159,25 @@ function TreeViewBranch(props: TreeViewBranchProps): ReactElement {
     );
     const [isActualLeafNode, setIsActualLeafNode] = useState<boolean>(props.isUserDefinedLeafNode || !props.children);
 
+    const updateTreeViewState = useCallback<TreeViewBranchContextProps["informParentIsLoading"]>(isLoading => {
+        if (!isLoading) {
+            setTreeViewState(treeViewState =>
+                treeViewState === TreeViewState.LOADING ? TreeViewState.EXPANDED : treeViewState
+            );
+        }
+    }, []);
+
     const toggleTreeViewContent = useCallback(() => {
         if (!isActualLeafNode) {
             setTreeViewState(treeViewState => {
-                if (
-                    treeViewState === TreeViewState.COLLAPSED_WITH_JS ||
-                    treeViewState === TreeViewState.COLLAPSED_WITH_CSS
-                ) {
+                if (treeViewState === TreeViewState.LOADING) {
+                    // TODO:
```

Comment:
Intentional?

Reply:
Yup, for now I have implemented it as "when loading and user clicks, ignore the click and stay loading". But it could also be that we want to reset the state. Will leave that for the feedback round.